### ISSUE TITLE
Missing "fib" modality in rig.json

### ISSUE THREAD

**Issue by hanhou** [AllenNeuralDynamics/dynamic-foraging-task#1056]
When I'm investigating another issue, I noticed that there are many sessions in docDB that have "fib" in `data_description.modality` but not in `rig.modalities`.  For example:

```json
{
    "_id": "625df1c2-1b38-4875-9a5f-09bbaca0e213",
    "data_description": {
      "name": "behavior_751152_2024-10-13_16-07-35",
      "modality": [
        {
          "name": "Behavior",
          "abbreviation": "behavior"
        },
        {
          "name": "Behavior videos",
          "abbreviation": "behavior-videos"
        },
        {
          "name": "Fiber photometry",
          "abbreviation": "fib"
        }
      ]
    },
    "rig": {
      "modalities": [
        {
          "name": "Behavior",
          "abbreviation": "behavior"
        },
        {
          "name": "Behavior videos",
          "abbreviation": "behavior-videos"
        }
      ]
    }
  },
  {
    "_id": "fbb5bb77-fef3-4f3b-93f2-560a8565488d",
    "data_description": {
      "name": "behavior_764787_2024-10-15_13-07-30",
      "modality": [
        {
          "name": "Behavior",
          "abbreviation": "behavior"
        },
        {
          "name": "Behavior videos",
          "abbreviation": "behavior-videos"
        },
        {
          "name": "Fiber photometry",
          "abbreviation": "fib"
        }
      ]
    },
    "rig": {
      "modalities": [
        {
          "name": "Behavior",
          "abbreviation": "behavior"
        },
        {
          "name": "Behavior videos",
          "abbreviation": "behavior-videos"
        }
      ]
    }
  }
```

[Here is the query ](https://api.allenneuraldynamics.org/v1/metadata_index/data_assets?filter={%22data_description.modality.abbreviation%22:%22fib%22,%22rig.modalities.abbreviation%22:{%22$ne%22:%22fib%22}}&projection={%22data_description.name%22:1,%20%22rig.modalities%22:1,%20%22data_description.modality%22:1,%20%22rig.rig_id%22:1}) showing all sessions with this problem. 

We should:
1. locate and fix the root cause, presumbly in how we generated `rig.json`. @XX-Yin @micahwoodard 
2. correct the metadata in docDB (shall we just directly add a patch to these records? how? @dyf)
3. In the long run, [a compatibility check](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/510#issuecomment-2489835716) should catch this kind of cross-validation problem.

- see also https://github.com/AllenNeuralDynamics/dynamic-foraging-task/issues/510

**Comment by alexpiet** [comment]
@hanhou Your query of sessions with this problem are exclusively from non-FIP rigs. The FIP rigs are 1D, 2D, 3D, 6D, 7D, 8D which do not show up in the log.

So the problem is with the data_description.json, not the rig.json

**Comment by XX-Yin** [comment]
@hanhou @alexpiet Previously, the session description was generated locally, and the code would exclude the **fib modality** if there was no fib data.  That functionality has been removed and replaced by online generation during upload, which may be the source of the problem.

**Comment by hanhou** [comment]
Interesting! Then all make sense to me. Here may be what happened:

1. Prior to [this PR](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/870), we incorrectly added "fib" to "data_description.modalities" of pure behavior sessions without FIP.
2. In the Streamlit app, to compare with pure behavior _and_ behavior + FIP sessions in my temporary pipeline, we should have queried "behavior" from `data_description.modalities`, but we actually used "fib". Interestingly, two wrongs make a right: because of 1, we actually queried all behavior-including sessions. So for a long time, sessions from both pipelines seemed to match in Streamlit.
3. Recently, after [the PR](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/870), point 1 was fixed and we started to see discrepancies in Streamlit.

**Comment by XX-Yin** [comment]
I speculate that the uploading process added the **fib modality** to the data description by default if there is a fib folder. 

@alexpiet If there is no fib data, have we already excluded the fib folder in the manifest file?

**Comment by hanhou** [comment]
> I speculate that the uploading process added the **fib modality** to the data description by default if there is a fib folder.
> 
> @alexpiet If there is no fib data, have we already excluded the fib folder in the manifest file?

Do you think this [PR](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/870) has fixed the problem?

**Comment by alexpiet** [comment]
Yes, my understanding is that PR fixed the problem. The modalities in data_description will be the modalities listed in the upload manifest. Spot checking actual manifests from this week, they correctly do NOT list FIP for behavior only sessions

**Comment by hanhou** [comment]
## The correct way of querying "FIP" sessions

Since the `data_description.modality` was unreliable, at this point, the correct query might be 

```python
{"session.data_streams.stream_modalities.abbreviation": "fib"}
```

To validate, for sessions which have "fib" as a stream, all of them have "fib" in "rig.modalities", if "rig" exists
https://api.allenneuraldynamics.org/v1/metadata_index/data_assets?filter={"session.data_streams.stream_modalities.abbreviation":"fib","rig":{"$ne":null},"rig.modalities.abbreviation":{"$ne":"fib"}}

@alexpiet @XX-Yin @rachelstephlee @hagikent  does this make sense?

- related to https://github.com/AllenNeuralDynamics/foraging-behavior-browser/issues/94

**Comment by rachelstephlee** [comment]
Discussed in kanban meeting 11/26

David says the fix forward has been done. 
David says he's volunteered for the fixes in the past. I will defer to @dyf  for how we can correct the queries.

**Comment by hanhou** [comment]
In this plot, the red circle should be inside the green circle and overlap perfectly with the blue circle.

   <img width="600" alt="image" src="https://github.com/user-attachments/assets/791899e1-a1ff-43ea-8f67-1590880d2ee4"/>

**Comment by dougollerenshaw** [comment]
@dyf is this still on your radar? Is this ultimately a metadata cleanup issue? Should it be a ticket in the data_migration_scripts repo that Dan or I can tackle?
